Sources/NIOOpenSSL/SSLContext.swift

```diff
@@ -108,12 +108,13 @@ private func alpnCallback(ssl: UnsafeMutablePointer<SSL>?,
 /// which are expected to be broadly the same.
 public final class SSLContext {
     private let sslContext: UnsafeMutablePointer<SSL_CTX>
+    private let passphraseUserData: UnsafeMutableRawPointer?
     internal let configuration: TLSConfiguration
 
     /// Initialize a context that will create multiple connections, all with the same
     /// configuration.
-    public init(configuration: TLSConfiguration) throws {
-        precondition(initialized)
+    internal init(configuration: TLSConfiguration, passphraseUserData: UnsafeMutableRawPointer?) throws {
```

Comment:
- let's take and store this `passphraseUserData` as `Unmanaged<thing>` and only make it an `UnsafeMutableRawPointer` when we actually need it
- also: the caller should pass us that thing however it likes, we should then `retain` it in `init` and `release` in `deinit`. It's non-standard to consume a reference (and even more odd to consume a reference that we obtain as `UnsafeMutableRawPointer`.

Reply:
I cannot take this as an Unmanaged without erasing its type by storing it in a protocol existential. If we're doing that, we may as well just take the protocol existential directly and create the unmanaged in the inner function.